### PR TITLE
Refactor dMin to dMax

### PR DESCRIPTION
**Pull request by mituritsyn**
Configurator part for https://github.com/betaflight/betaflight/pull/13908

### CODE REVIEW

**Comment by ctzsnooze** [comment]
Maybe make the 'D' in 'Derivative' label bold, also?

**Comment by ctzsnooze** [comment]
Also, in image above, `D Max` values are shown as being smaller than Derivative values, whereas they should always be greater.  Perhaps update the image ?

**Comment by haslinghuis** [comment]
Backwards compatibility

**Comment by mituritsyn** [comment]
> i'd suggest if/then to swap labels only irregardless of other logic.

I'm not sure I've made the best implementation.

https://github.com/betaflight/betaflight-configurator/pull/4173/commits/2b30712c3f5021ffad5fbd8efa685fb3a05219e2

**Comment by haslinghuis** [comment]
- Normally only have to change english locale only - other locales get overwritten by crowdin but it does not hurt to have them updated.
- Backwards compatibilty looks to be resolved except for the sliderWarning
- In PR with 4.5.2:
image
- In master with 4.5.2:
Screenshot from 2024-09-22 16-17-40

File: src/js/TuningSliders.js [hunk 1]
```diff
@@ -241,10 +241,10 @@ TuningSliders.dtermFilterSliderDisable = function() {
 
 TuningSliders.updateSlidersWarning = function(slidersUnavailable = false) {
     const WARNING_P_GAIN = 70;
-    const WARNING_DMAX_GAIN = 60;
+    const WARNING_D_MAX_GAIN = 60;
     const WARNING_I_GAIN = 2.5 * FC.PIDS[0][0];
-    const WARNING_DMIN_GAIN = 42;
-    const enableWarning = FC.PIDS[0][0] > WARNING_P_GAIN || FC.PIDS[0][1] > WARNING_I_GAIN || FC.PIDS[0][2] > WARNING_DMAX_GAIN || FC.ADVANCED_TUNING.dMinRoll > WARNING_DMIN_GAIN;
+    const WARNING_D_GAIN = 42;
+    const enableWarning = FC.PIDS[0][0] > WARNING_P_GAIN || FC.PIDS[0][1] > WARNING_I_GAIN || FC.PIDS[0][2] > WARNING_D_GAIN || FC.ADVANCED_TUNING.dMaxRoll > WARNING_D_MAX_GAIN;
```
Comment (haslinghuis) [inline src/js/TuningSliders.js:247]: Perhaps (need to import semver)

```suggestion
        const enableWarning = semver.lt(FC.CONFIG.apiVersion, '1.47.0')
        ? FC.PIDS[0][0] > WARNING_P_GAIN || FC.PIDS[0][1] > WARNING_I_GAIN || FC.PIDS[0][2] > WARNING_D_MAX_GAIN || FC.ADVANCED_TUNING.dMaxRoll > WARNING_D_GAIN
        : FC.PIDS[0][0] > WARNING_P_GAIN || FC.PIDS[0][1] > WARNING_I_GAIN || FC.PIDS[0][2] > WARNING_D_GAIN || FC.ADVANCED_TUNING.dMaxRoll > WARNING_D_MAX_GAIN;
```

**Comment by mituritsyn** [comment]
> * Normally only have to change english locale only

nice to know)
there is duplicate pidTuningDerivativeHelp key, I'm sure there should be only one but they have similar text.

**Comment by haslinghuis** [comment]
**Last edition** [2672](https://github.com/betaflight/betaflight-configurator/commit/042211b187916d657a50e41bf3c374e35fb4b5f7):

`Controls the strength of dampening to ANY motion on the craft.  For stick moves, the D-term dampens the command. For an outside influence (prop wash OR wind gust) the D-term dampens the influence.<br><br>Higher gains provide more dampening and reduce overshoot by P-term and FF.<br>However, the D-term is VERY sensitive to gyro high frequency vibrations (noise | magnifies by 10x to 100x).<br><br>High frequency noise can cause motor heat and burn out motors if D-gains are too high or the gyro noise is not filtered well (see Filters tab).<br><br>Think of the D-term as the shock absorber on your car, but with the negative inherent property of magnifying high frequency gyro noise.


**Previous edition** [2572](https://github.com/betaflight/betaflight-configurator/commit/1dbe298abd0e9c2a9018d582c1f662b4e08f0a73) line 1861 should probably be removed - BUT seems currently in use:

Baseline damping of ANY motion of the craft. <br /><br />Opposes movement whether caused by stick inputs or external influences (e.g. prop-wash or wind gusts)<br /><br />Higher D gains provide more stability and reduce overshoot.<br /><br />D amplifies noise (magnifies by 10x to 100x).  This can burn out motors if gains are too high or D isn't filtered well.<br /><br />D-term is a bit like the shock absorber on your car.
`

**Comment by haslinghuis** [comment]
@mituritsyn dmax calculation is still off

PR with `13908`:

image

Master:

image

**Review by haslinghuis — APPROVED** [review]

**Comment by mituritsyn** [comment]
> @mituritsyn
> 
> * noticed `D_MAX` calculation has changed (an improvement as `D_MAX` is allowed below `D` while `D_MIN` could not be lower than `D`) as we no longer use `dminRatio` in firmware calc.

No, I'm trying to keep the old behaviour. I've updated the calculation formula in the firmware.

**Pull request merged**